File: multi/src/main/java/com/wyyu/multi/cell/ClassCellManager.java
package com.wyyu.multi.cell;

import androidx.annotation.NonNull;

/**
 * Created by wyyu on 2019-09-27.
 *
 * 已列表中数据类型为唯一标识的 CellManager
 **/

public class ClassCellManager extends AbsCellManager<Class<?>, Object> {

    @Override public Class<?>[] loadKeyArray() {
        return new Class[DEFAULT_LENGTH];
    }

    @Override public Class<?> loadKeyFromItem(@NonNull Object item) {
        return item.getClass();
    }
}

File: multi/src/main/java/com/wyyu/multi/cell/ClassCellManager.kt
package com.wyyu.multi.cell

/**
 * Created by wyyu on 2021/7/29.
 **/

class ClassCellManager : AbsCellManager<Class<out Any>, Any>() {

    override fun loadKeyFromData(itemData: Any): Class<out Any> {
        return itemData::class.java
    }
}